internal/pkg/agent/application/paths/paths_test.go

```diff
@@ -92,3 +94,123 @@ func TestHasPrefixWindows(t *testing.T) {
 		})
 	}
 }
+
+func TestResolveControlSocketWithInstalledState(t *testing.T) {
```

Comment:
Thank you, I appreciate that you added a unit test.

So, we only have one case where `ResolveControlSocketWithInstalledState` returns a different result than `ControlSocketFromPath` and it's for Windows when the agent is detected as installed...

I was more thinking about having a small UT for [ResolveControlSocket()](https://github.com/elastic/elastic-agent/pull/5879/files#diff-8260d5183b15bf6f7e77073338ec4994ce3a058aa618b4b015993f2f56b18474R67) where we can prepare a temp directory to be the top path with and without the `.installed` marker and having explicit expected results.

Reply:
i understand what you're trying to achieve.
i'd rather address this in a followup as we may hit the window for next release with actual state of the PR and tests which should be good enough to get us covered.

Reply:
`ResolveControlSocket()` is only called from a single place, and that code already calls `paths.RunningInstalled()` right before `ResolveControlSocket()`.

https://github.com/michalpristas/elastic-agent/blob/be8773ff18061ec0a7570fa43db902ff86e709fe/internal/pkg/agent/cmd/run.go#L713-L736

So IMO `ResolveControlSocket` just should always take whether we know we are installed as an argument and we can avoid having two functions. Then having the installed detection is never a part of this functions job, so we don't need to test that part here at all.

Just reading the test cases here it is really hard to get a picture of what is supposed to happen, fundamentally:
1. When on Windows, whether you are installed or not changes the result.
2. On Darwin and Linux, being installed makes no difference.

It's not clear why we have 3 test cases for each platform to me, when there are naturally only 2.

It feels like the tests could be simpler.

P.S. 8.16 is likely going to be delayed and there will be another BC sometime next week. If there isn't, this PR has already missed the release. So let's not worry about landing this and instead focus on whether we are happy with it as that is already out of our control until we know the next BC date.

Reply:
will update test just a note we have 4 test cases per each platform not 3. 
they are following matrix of values installed vs control socket differs to expected
- installed same paths,
- installed different paths,
- not installed same paths
- not installed different paths

what we track is if control socket persists the same except for windows use case when it's installed and paths are different when we fix the path to be `npipe:///elastic-agent-system`

Reply:
test logic stayed the same, hope that change for test to be more matrix like will make it more clear/readable
```go
		{"darwin", false, false, false, ""},
		{"darwin", true, false, false, ""},
		{"darwin", false, true, false, ""},
		{"darwin", true, true, false, ""},

		{"linux", false, false, false, ""},
		{"linux", true, false, false, ""},
		{"linux", false, true, false, ""},
		{"linux", true, true, false, ""},

		{"windows", false, false, false, ""},
		{"windows", true, false, false, ""},
		{"windows", false, true, false, ""},
		{"windows", true, true, false, WindowsControlSocketInstalledPath},
```